src/vmm/src/config.rs

```diff
@@ -1,13 +1,100 @@
 // Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
 
+use std::collections::HashMap;
 use std::convert::TryFrom;
 use std::fmt;
+use std::num;
 use std::path::PathBuf;
 use std::result;
 
 use super::{DEFAULT_HIGH_RAM_START, DEFAULT_KERNEL_CMDLINE};
 
+trait ArgParsing {
+    fn parse(param_name: &'static str, raw: String) -> Result<Self, ParsingError>
+    where
+        Self: Sized;
+}
+
+macro_rules! argparsing{
+    ($($t:ty)*) => ($(impl ArgParsing for $t {
+        fn parse(param_name: &'static str, raw: String) -> Result<Self, ParsingError> {
+            raw.parse().map_err(|err: num::ParseIntError| ParsingError::ParsingFailed(param_name, err.to_string()))
+        }
+    })*)
+}
+
+argparsing! { u32 u64 num::NonZeroU8 }
+
+impl ArgParsing for PathBuf {
+    fn parse(_param_name: &'static str, raw: String) -> Result<Self, ParsingError> {
+        Ok(PathBuf::from(raw))
+    }
+}
+impl ArgParsing for String {
+    fn parse(_param_name: &'static str, raw: String) -> Result<Self, ParsingError> {
+        Ok(raw)
+    }
+}
+
+fn tokenize(input: String) -> HashMap<String, String> {
```

Comment:
Also, when I think a bit more about it, why not fail early here? `eg.` if you find something like `path=` or `path`, return an error here. Or may be not. Because adding that logic here would complicate handling values without `=` if they ever will be required later on. So perhaps the way it is is fine.

Reply:
That's exactly what I had in mind. This function should just tokenize the input string by `=`, and this operation should always succeed IMO. If a required parameter is missing the conversion will eventually fail anyway.

Also, as you correctly mentioned, not failing early allows to quickly add support for flags, e.g. `--path` (if they will ever be needed).

Finally, I've renamed the function to `cfg_args_to_map`. I don't see any problem with a tokenize function returning an HashMap TBH, but I don't have a strong opinion on this. Thanks for all your comments so far!